pkg/relay/processmanager.go

```diff
@@ -0,0 +1,178 @@
+package relay
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/blocknative/dreamboat/pkg/structs"
+	"github.com/flashbots/go-boost-utils/bls"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	ResponseQueueSubmit = iota
+	ResponseQueueRegister
+	ResponseQueueOther
+)
+
+type ProcessManager struct {
+	LastRegTime map[string]uint64 // [pubkey]timestamp
+	lrtl        sync.RWMutex      // LastRegTime RWLock
+
+	VerifySubmitBlockCh       chan VerifyReq
+	VerifyRegisterValidatorCh chan VerifyReq
+	VerifyOtherCh             chan VerifyReq
+
+	StoreCh chan StoreReq
+
+	m ProcessManagerMetrics
+}
+
+func NewProcessManager(verifySize, storeSize uint) *ProcessManager {
+	rm := &ProcessManager{
+		LastRegTime: make(map[string]uint64),
+
+		VerifySubmitBlockCh:       make(chan VerifyReq, verifySize),
+		VerifyRegisterValidatorCh: make(chan VerifyReq, verifySize),
+		VerifyOtherCh:             make(chan VerifyReq, verifySize),
+
+		StoreCh: make(chan StoreReq, storeSize),
+	}
+	rm.initMetrics()
+	return rm
+}
+
+func (rm *ProcessManager) RunVerify(num uint) {
+	for i := uint(0); i < num; i++ {
+		go rm.VerifyParallel()
+	}
+}
+
+func (rm *ProcessManager) RunStore(store Datastore, ttl time.Duration, num uint) {
+	for i := uint(0); i < num; i++ {
+		go rm.ParallelStore(store, ttl)
+	}
+}
+
+func (rm *ProcessManager) RunCleanup(checkinterval uint64, cleanupInterval time.Duration) {
+	for {
+		now := uint64(time.Now().Unix())
+		var keys []string
+		rm.lrtl.RLock()
```

Comment:
Can we put both of these inner for-loops in subfunctions and use `defer` to unlock the mutex?  I am extremely paranoid about panics while holding locks, since HTTP servers will often recover from panicking requests in middleware and subsequently deadlock.

Reply:
It's simple append here but well, as you wish... :)

Reply:
done

Reply:
Yeah, I hesitated to even mention it, but it helps to not stop and check while re-reading the code.  So if nothing else, I think it's helpful to signal quite loudly that the mutex behavior is correct.  🤷‍♂️ 

Anyway, thanks :)